src/Microsoft.Identity.Client/Internal/Requests/SilentRequest.cs

```diff
@@ -58,49 +63,68 @@ internal override async Task<AuthenticationResult> ExecuteAsync(CancellationToke
                     "Token cache is set to null. Silent requests cannot be executed.");
             }
 
-            MsalAccessTokenCacheItem msalAccessTokenItem = null;
-
-            // Look for access token
-            if (!ForceRefresh)
+            if (!String.IsNullOrWhiteSpace(UserProvidedRefreshToken))
```

Comment:
This if block here basically makes SilentRequest into two different request behaviors.  Why not just create a new ByRefreshTokenRequest so that you don't need this conditional logic here and possibly elsewhere and just create a speicfic request object to handle this case.  I think you could then remove UserProvidedRefreshToken from the RequestParameters so that concept doesn't have to leak to other implementations since it's only relevant for this specific request type.

Reply:
Yes, good point. I would also move the code into a separate folder under the Features instead of under ConfidentialClient, if that makes sense.